server/metrics/metrics.go

```diff
@@ -596,51 +588,54 @@ func (m *Metrics) createEndpoint() (*http.ServeMux, error) {
 
 func (m *Metrics) startEndpoint(
 	stop chan struct{}, network, address string, me http.Handler,
-) error {
-	l, err := net.Listen(network, address)
-	if err != nil {
-		return fmt.Errorf("creating listener: %w", err)
-	}
+) {
+	stopped := false
+	for !stopped {
+		l, err := net.Listen(network, address)
+		if err != nil {
+			logrus.Warnf("Failed to create listener %v", err)
+			time.Sleep(15 * time.Second)
+			continue
+		}
+
+		srv := &http.Server{
+			Handler: me,
+		}
+
+		go func() {
+			<-stop
+			stopped = true
```

Comment:
should this be after the if statement?  If the shutdown fails, you're not stopped are you?